crates/rethnet_rpc_server/src/lib.rs

```diff
@@ -0,0 +1,455 @@
+                                ),
+                                MethodInvocation::Hardhat(HardhatMethodInvocation::SetBalance(
+                                    address,
+                                    balance,
+                                )) => {
+                                    response(id, handle_set_balance(state, address, balance).await)
+                                }
+                                MethodInvocation::Hardhat(HardhatMethodInvocation::SetCode(
+                                    address,
+                                    code,
+                                )) => response(id, handle_set_code(state, address, code).await),
+                                MethodInvocation::Hardhat(HardhatMethodInvocation::SetNonce(
+                                    address,
+                                    nonce,
+                                )) => response(id, handle_set_nonce(state, address, nonce).await),
+                                MethodInvocation::Hardhat(
+                                    HardhatMethodInvocation::SetStorageAt(address, position, value),
+                                ) => response(
+                                    id,
+                                    handle_set_storage_at(state, address, position, value).await,
+                                ),
+                                // TODO: after adding all the methods here, eliminate this
+                                // catch-all match arm:
+                                _ => response(
+                                    id,
+                                    ResponseData::<serde_json::Value>::new_error(
+                                        -32601,
+                                        "Method not found",
+                                        match serde_json::to_value(method) {
+                                            Ok(value) => Some(value),
+                                            Err(_) => None,
+                                        }
+                                    )
+                                ),
+                            }
+                        }
+                    }
+                },
+            ),
+        )
+        .with_state(state)
+}
+
+/// accepts an address to listen on (which could be a wildcard like 0.0.0.0:0), and a set of
+/// initial accounts to initialize the state.
+/// returns the address that the server is listening on (with any input wildcards resolved).
+pub async fn run(
+    address: SocketAddr,
+    config: RpcHardhatNetworkConfig,
+    genesis_accounts: HashMap<Address, AccountInfo>,
+) -> Result<SocketAddr, std::io::Error> {
+    let listener = TcpListener::bind(address)?;
+    let address = listener.local_addr()?;
+
+    let rethnet_state: StateType = if let Some(config) = config.forking {
+        Arc::new(RwLock::new(Box::new(ForkState::new(
```

Comment:
Consider splitting this into multiple lines with variable assignment to make it easier to read.

```rust
let runtime = tokio::runtime::Builder::new_multi_thread()
                    .enable_io()
                    .enable_time()
                    .build()
                    .expect("failed to construct async runtime");

let generator = { .. };
let state = { .. };

// etc.
```

Reply:
thanks, i like that. addressed in https://github.com/NomicFoundation/hardhat/pull/3998/commits/ebeedf15fb102b05d7f642a4d4f2b1d20392fc92